script.js

```diff
@@ -73,159 +73,201 @@ if (
     };
   });
 
-  //------ Rating filter -------
+  //--------- FILTER BY RATING ----------
   const ratingInput = document.getElementsByName("rating");
 
-  function ratingFilter() {
-    let cards = document.querySelectorAll(".challenge-item");
-    for (let i = 0; i < cards.length; i++) {
-      if (
-        cards[i].querySelector("ul.rating").ariaValueNow >=
-          current_star_level_from &&
-        cards[i].querySelector("ul.rating").ariaValueNow <=
-          current_star_level_to
-      ) {
-        cards[i].classList.remove("is-hidden");
-      } else {
-        cards[i].classList.add("is-hidden");
+  class RatingFilter {
```

Comment:
Although a class (which is very associated with OOP), the code in this class interacts directly with global variables (i.e. the DOM via `document`), which means that it breaks one of the core concepts of OOP/OOD, which is "encapsulation".

This means that the code is not really object-oriented, even if it uses a class.